agent/agent_endpoint.go

```diff
@@ -86,7 +88,22 @@ func (s *HTTPServer) AgentMetrics(resp http.ResponseWriter, req *http.Request) (
 	if rule != nil && !rule.AgentRead(s.agent.config.NodeName) {
 		return nil, acl.ErrPermissionDenied
 	}
+	if format := req.URL.Query().Get("format"); format == "prometheus" {
```

Comment:
What usually happens in other systems is that the content-type header gets used for format negotiation. Unfortunately prometheus content-type is quite loosy: `text/plain; version=0.0.4; charset=utf-8`. But that could still be used to auto-guess.

Reply:
True, but since anyway we are not using the default endpoint /metrics, it will require configuration on prometheus side anyway, soI just kept the suggestion from @mkeeler https://github.com/hashicorp/consul/issues/4014#issuecomment-379034920 to implement it the same way as nomad